src/Controller/TagsController.php

```diff
@@ -246,4 +246,29 @@ public function search()
             $search
         ]);
     }
+
+    /**
+     * Display list of tags for autocompletion.
+     *
+     * @param String $filter Filters the tags list with only those that contain the
+     *                       search string.
+     */
+    public function autocomplete($search)
+    {
+        $this->helpers[] = 'Tags';
+
+        $query = $this->Tags->find();
+        $query->select(['name', 'id', 'nbrOfSentences']);
+
+        if (!empty($search)) {
+            $query->where(['name LiKE' => "$search%"]);
```

Comment:
This approach may allow SQL injections. For example, I can type `%VO` and get results such as `SVO`. (CakePHP does some kind of quote escaping, but let’s not solely rely on that.) Have a look at `src/Command/FixNewUsersTOSSettingCommand.php` to see an example of how to implement it using binded variables.

Reply:
Thank you for pointing that out! Binding was not enough to prevent "%" from being interpreted, so I had to work it around by replacing and escaping "|" characters. Doing so assumes "|" is not used in tags.

Reply:
I see. Sorry, my comment about the percent was misleading. Using bind() alone protects from SQL injection, but the percent problem is a different thing, that is not a security threat. It’s just that using percent will have special meaning that it not what the user expects. But you are welcome to escape the percent too.

I think you are in the right direction, but the ESCAPE operator should be used to set the escape character (which is backslash by default), not the characters you actually want to escape. So if ESCAPE is backslash, you need to write '\\%' in the LIKE pattern to match a percent. If a tag name is, say, '100% correct', then if the user types '100%', we want that name to show up. In addition, we want the tag '100 words' NOT to show up. Finally, you want to escape the underscore because it’s also a metacharacter (which means "any character"), and the escape character itself.

So maybe something like that:
```php
$pattern = str_replace(['\\', '%', '_'], ['\\\\', '\%', '\_'], $search).'%';
$query->where(['name LIKE :search'])->bind(':search', $pattern, 'string');
```

Reply:
I finally understand, I'm sorry for causing you so much trouble!